pumpkin-lib/src/engine/sat/assignments_propositional.rs

```diff
@@ -221,12 +223,15 @@ impl AssignmentsPropositional {
         self.assignment_info[variable] = PropositionalAssignmentInfo::Unassigned;
     }
 
+    /// Enqueues a [`Literal`] which was the result of a decision by a [`Brancher`].
     pub fn enqueue_decision_literal(&mut self, decision_literal: Literal) {
         pumpkin_assert_simple!(!self.is_literal_assigned(decision_literal));
 
         let _ = self.make_assignment(decision_literal, ConstraintReference::NULL);
     }
 
+    /// Enqueues a propagated [`Literal`]; i.e. a [`Literal`] which was determined to be true by the
+    /// logic of a [`Propagator`].
```

Comment:
Not necessarily true. The clausal propagator does not implement `Propagator`. I would say something along the lines: `i.e. a literal which is true because of a specified reason`.